### PR TITLE
CIRC-1785 Add service point for the effective location

### PR DESCRIPTION
**Pull request by rgupta-nla**
## Purpose
https://issues.folio.org/browse/CIRC-1785
Add "service point for the effective location" as staff slip token

Purpose
Purpose of this PR is to add "service point for the effective location" as staff slip token. The changes would reflect in staff slip produced in the Requests app (pick slip) and staff slip produced in the Check in app (hold, delivery request and transfer slips).

## Approach
We are getting ServicePoint object information by fetching it passing the servicePoingId. We then grab the service point location (primary location ) and stick it to the item context. This is done for pick slip resource.

For the checkin app, the location name is already available in the Item object so we just add it to json object.

#### TODOS and Open Questions
- [ ] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 1]
```diff
@@ -108,6 +109,29 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+      log.info("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: {}, primaryServicePoint = null", entries);
+      return entries;
+    }
+
+    if (entries == null) {
+      log.info("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: null, primaryServicePointName {}", primaryServicePoint.getName());
+      return new JsonObject();
+    }
+
+    log.debug("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: {}, primaryServicePointName = {}", entries, primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:123]: it should be at the beginning of the method

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 2]
```diff
@@ -108,6 +109,29 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+
+    entries.getJsonArray(PICK_SLIPS_KEY)
+      .stream()
+      .map(JsonObject.class::cast)
+      .map(pickSlip -> pickSlip.getJsonObject(ITEM))
+      .forEach(item -> item.put("effectiveLocationPrimaryServicePointName", primaryServicePoint.getName()));
+
+    log.debug("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: {}, primaryServicePointName = {}", entries, primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:131]: You don't need to log the parameters multiple times. Instead, please log the result using the following format:

LOGGER.info("sumWithCondition:: result: {}", sumResult);

return sumResult;

You can refer to the documentation for logging at https://wiki.folio.org/display/DD/Logging.

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 3]
```diff
@@ -108,6 +109,29 @@ public static JsonObject createStaffSlipContext(Request request) {
     return createStaffSlipContext(request.getItem(), request);
   }
 
+  public static JsonObject addPrimaryServicePointNameToStaffSlipContext(JsonObject entries, ServicePoint primaryServicePoint) {
+    if (primaryServicePoint == null) {
+      log.info("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: {}, primaryServicePoint = null", entries);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:114]: Please consider updating the message in the code to accurately reflect the executed condition

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 4]
```diff
@@ -108,6 +109,29 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+      return entries;
+    }
+
+    if (entries == null) {
+      log.info("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: null, primaryServicePointName {}", primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:119]: Please consider updating the message in the code to accurately reflect the executed condition

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 5]
```diff
@@ -108,6 +109,30 @@ public static JsonObject createStaffSlipContext(Request request) {
     return createStaffSlipContext(request.getItem(), request);
   }
 
+  public static JsonObject addPrimaryServicePointNameToStaffSlipContext(JsonObject entries, ServicePoint primaryServicePoint) {
+    log.info("addPrimaryServicePointNameToStaffSlipContext:: parameters entries: {}", entries);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:113]: The log should be set to debug level and include the second parameter in its message

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 6]
```diff
@@ -108,6 +109,30 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+    if (primaryServicePoint == null) {
+      log.info("primaryServicePoint object is null");
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:115]: Please add the name of the method to the log message

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 7]
```diff
@@ -108,6 +109,30 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+      return entries;
+    }
+
+    if (entries == null) {
+      log.info("entries JsonObject is null, primaryServicePointName: {}", primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:120]: Please add the name of the method to the log message

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 8]
```diff
@@ -108,6 +109,30 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+      return new JsonObject();
+    }
+
+    log.debug("addPrimaryServicePointNameToStaffSlipContext:: primaryServicePointName: {}", primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:124]: it appears to be redundant

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 9]
```diff
@@ -108,6 +109,30 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+
+    entries.getJsonArray(PICK_SLIPS_KEY)
+      .stream()
+      .map(JsonObject.class::cast)
+      .map(pickSlip -> pickSlip.getJsonObject(ITEM))
+      .forEach(item -> item.put("effectiveLocationPrimaryServicePointName", primaryServicePoint.getName()));
+
+    log.debug("Result after adding Primary Service Point Name to Staff Skip Json:: entries: {}", entries);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:132]: Please add the name of the method to the log message

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java (hunk 8 above)
Comment (rgupta-nla) [reply src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:124]: This prints primaryServicePointName (this is an important field that we want to make sure is shown in logs)

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 10]
```diff
@@ -108,6 +109,29 @@ public static JsonObject createStaffSlipContext(Request request) {
...
+      return new JsonObject();
+    }
+
+    entries.getJsonArray(PICK_SLIPS_KEY)
+      .stream()
+      .map(JsonObject.class::cast)
+      .map(pickSlip -> pickSlip.getJsonObject(ITEM))
+      .forEach(item -> item.put("effectiveLocationPrimaryServicePointName", primaryServicePoint.getName()));
+
+    log.debug("addPrimaryServicePointNameToStaffSlipContext:: Result after adding Primary Service Point Name to Staff Slip Json:: entries: {}, primaryServicePointName: {}", entries, primaryServicePoint.getName());
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java:130]: please change log level to info and additionally, consider simplifying the log message to"addPrimaryServicePointNameToStaffSlipContext:: result entries: {}, primaryServicePointName: {}", entries, primaryServicePoint.getName()

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**